modules/globebrowsing/tile/asynctiledataprovider.h

```diff
@@ -25,39 +25,100 @@
 #ifndef __OPENSPACE_MODULE_GLOBEBROWSING___ASYNC_TILE_DATAPROVIDER___H__
 #define __OPENSPACE_MODULE_GLOBEBROWSING___ASYNC_TILE_DATAPROVIDER___H__
 
-#include <modules/globebrowsing/other/concurrentjobmanager.h>
+
+#include <modules/globebrowsing/globebrowsingmodule.h>
+
+#include <modules/globebrowsing/other/prioritizingconcurrentjobmanager.h>
+#include <modules/globebrowsing/other/pixelbuffercontainer.h>
 #include <modules/globebrowsing/tile/tileindex.h>
 
+#include <ghoul/opengl/ghoul_gl.h>
+
+#include <map>
+#include <set>
 #include <unordered_map>
 
 namespace openspace {
 namespace globebrowsing {
     
+//class GlobeBrowsingModule;
 struct RawTile;
 class RawTileDataReader;
 
+/**
+ * The responsibility of this class is to enqueue tile requests and fetching finished
+ * <code>RawTile</code>s that has been asynchronously loaded.
+ */
 class AsyncTileDataProvider {
 public:
-    AsyncTileDataProvider(std::shared_ptr<RawTileDataReader> textureDataProvider,
-        std::shared_ptr<ThreadPool> pool);
+    /**
+     * \param textureDataProvider is the reader that will be used for the asynchronos
+     * tile loading.
+     */
+    AsyncTileDataProvider(std::shared_ptr<RawTileDataReader> textureDataProvider);
+
+    /**
+     * Creates a job which asynchronously loads a raw tile. This job is enqueued.
+     */
+    bool enqueueTileIO(const TileIndex& tileIndex);
 
-    bool enqueueTileIO(const TileIndex& tileIndex);        
+    /**
+     * Get all finished jobs.
+     */
     std::vector<std::shared_ptr<RawTile>> getRawTiles();
+
+    /**
+     * Get one finished job.
+     */
     std::shared_ptr<RawTile> popFinishedRawTile();
 
+    void update();
     void reset();
-    void clearRequestQueue();
 
     std::shared_ptr<RawTileDataReader> getRawTileDataReader() const;
     float noDataValueAsFloat() const;
 
 protected:
-    virtual bool satisfiesEnqueueCriteria(const TileIndex&) const;
+    enum class ResetRawTileDataReader {
```

Comment:
If you include `ghoul/misc/boolean.h` you can use `using ResetRawTileDataReader = ghoul::Boolean;`